app/models/email_domain_block.rb

```diff
@@ -28,6 +28,8 @@ class EmailDomainBlock < ApplicationRecord
 
   validates :domain, presence: true, uniqueness: true, domain: true
 
+  scope :parents, -> { where(parent_id: nil) }
```

Comment:
Really makes you contemplate the whole circle of life, doesnt it? Could there even be a parent who does not themselves also have a parent? Much to consider.